ios-contact-manager/ios-contact-manager/Controller/ContactEditViewController.swift

```diff
@@ -7,28 +7,36 @@
 
 import UIKit
 
-final class ContactEditViewController: ContactCreateViewController {
-    override func configureViewComponents() {
-        self.ageTextField.keyboardType = .numberPad
-        self.phoneNumberTextField.keyboardType = .phonePad
-        
-        if let contact = contact {
-            super.contact = contact
-            self.nameTextField.text = contact.name
-            self.ageTextField.text = "\(contact.age)"
-            self.phoneNumberTextField.text = contact.phoneNumber
-        }
+final class ContactEditViewController: ContactFormViewController {
+    private let temporaryContainer: TemporaryContainer
+    
+    init?(coder: NSCoder, contact: Contact) {
+        self.temporaryContainer = TemporaryContainer(contact: contact)
+        super.init(coder: coder)
     }
     
+    required init?(coder: NSCoder) {
+        fatalError("Edit View Controller 생성 실패")
+    }
+    
+    override func configureViewComponents() {
+        guard let contact = self.temporaryContainer.contact else { return }
+        self.nameTextField.text = contact.name
+        self.ageTextField.text = "\(contact.age)"
+        self.phoneNumberTextField.text = contact.phoneNumber
+        super.configureViewComponents()
+    }
```

Comment:
```suggestion
    override func configureViewComponents() {
            super.configureViewComponents()
    
        guard let contact = self.temporaryContainer.contact else { return }
        self.nameTextField.text = contact.name
        self.ageTextField.text = "\(contact.age)"
        self.phoneNumberTextField.text = contact.phoneNumber
    }
```
=> 메서드에서 super 호출은 먼저하는게 맞습니다.

Reply:
하위 클래스의 이니셜라이저에서 `super.init()` 을 먼저 불렀을 때 에러가 발생한 경험이 있었습니다.
그것 때문에 super를 부르는건 항상 나중에 해야한다고 알고 있었는데, 생각해보니 `viewDidLoad()` 메서드의 경우에서도 `super.viewDidLoad()`를 먼저 하고 있었네요.
이니셜라이저의 경우는 현재 호출된 곳의 초기화 작업이 선행되어야 하기 때문에 super의 초기화는 나중에 진행하는 것이고,
메서드에서 super호출은 먼저해서 상위 개체의 오버라이딩 메서드를 먼저 준비 시키는 것이라고 이해하면 적절할까요?

Reply:
네 맞습니다 👍